common/inc/sfpu/ckernel_sfpu_add_int32.h

```diff
@@ -16,28 +16,47 @@ namespace ckernel
 namespace sfpu
 {
 
-template <bool APPROXIMATION_MODE, int ITERATIONS>
+template <bool APPROXIMATION_MODE, bool SIGN_MAGNITUDE_FORMAT, int ITERATIONS>
 inline void _add_int32_(const uint dst_offset) {
     // Operand A is input1 (int32)
     // Operand B is input2 (int32)
     // Output is int32
 
-    // Modifies LOAD/STORE to work with INT32 2's complement. Does not convert to INT32 sign-magnitude.
+    // Modifies LOAD/STORE to do INT32 sign-magnitude to 2's complement conversion, however
+    // in Blackhole this has no effect and format remains in original format.
     constexpr auto INSTR_MOD_LOAD_STORE = InstrModLoadStore::INT32_2S_COMP;
```

Comment:
Should this also depend on SIGN_MAGNITUDE_FORMAT flag in principle? I know in BH it doesn't have any effect, but you showed it matters in WH.

Reply:
Yes it should, I was told that InstrModLoadStore::INT32 was removed so this is the only option now.